src/modules/Auth/pages/RecoverPassword/RecoverPassword.tsx

```diff
@@ -0,0 +1,62 @@
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { Trans } from 'react-i18next';
+import { useEffect, useState } from 'react';
+import { Box } from '@mui/material';
+
+import { RecoverForm } from 'modules/Auth/features/RecoverPassword';
+import { recoveryLinkHealthCheckApi } from 'api';
+import { StyledBodyLarge, StyledLinkBtn } from 'shared/styles';
+import { Spinner } from 'shared/components';
+import { page } from 'resources';
+
+export const RecoverPassword = () => {
+  const [searchParams] = useSearchParams();
+
+  const navigate = useNavigate();
+
+  const key = searchParams.get('key');
+  const email = searchParams.get('email');
+
+  const [isLoading, setIsLoading] = useState<boolean>(true);
```

Comment:
please consider to use `useAsync`, I believe this will get rid of the local use of `loading` and `error` states

Reply:
I started out with `useAsync`, but I would still end up having to call the `execute` function inside a `useEffect`. I moved away from it because this approach is simpler